infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriter.java

```diff
@@ -38,12 +40,23 @@
                     .setNameFormat("LogUnit-Write-Processor-%d")
                     .build());
 
+    /**
+     * Barrier epoch persisted by the log unit.
+     */
+    private volatile long barrierEpoch;
+
+    /**
+     * Writer service shutdown wait duration.
+     */
+    private static final long WRITER_SERVICE_SHUTDOWN_TIMER = 5000;
```

Comment:
Would it be possible to move this CorfuRuntimeParameter (or parameter file)? (to avoid moving toward having constants/parameters reside in separate files)

Reply:
This is a server-side parameter. It would be out of place to put this in the client parameters.
It would be better (cleaner) to have a server-side parameters file where all these durations and limits can be stored.
Let me know if you still feel this should go in Runtime Parameters, I can make the change.

Reply:
Agreed, Runtime Parameter is not the natural place for this. My point was it's nice to have this important configurational parameters centralized in one location rather than spreading them in different files. However I adding a server side parameter file is outside the scope of this PR. I added an issue (#1356) so that we can address it on its own merit.

Reply:
probably want to use public static final Duration SHUTDOWN_TIMER = Duration.ofSeconds(5); from the server context